C++ API parity: Sigmoid

pbelevich opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27309 C++ API parity: MultiheadAttention
* #27382 C++ API parity: Linear
* #27538 C++ API parity: Threshold
* #27537 C++ API parity: Tanhshrink
* #27536 C++ API parity: Tanh
* #27535 C++ API parity: Softsign
* #27534 C++ API parity: Softshrink
* #27489 C++ API parity: Softplus
* **#27488 C++ API parity: Sigmoid**

Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

pbelevich committed:
C++ API parity: Sigmoid

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"




[ghstack-poisoned]

yf225 commented on torch/csrc/api/include/torch/nn/functional/activation.h:
```diff
@@ -74,6 +74,10 @@ inline Tensor celu(Tensor& input, const CELUOptions& options = {}) {
 
+inline Tensor sigmoid(const Tensor& input) {
+  return torch::sigmoid(input);
+}
```
We can probably remove `F::sigmoid`, because the Python version mentions that `nn.functional.sigmoid is deprecated. Use torch.sigmoid instead.`

yf225 commented on torch/csrc/api/src/nn/modules/activation.cpp:
```diff
@@ -189,5 +189,17 @@ void CELUImpl::pretty_print(std::ostream& stream) const {
+// ============================================================================
+
+Tensor SigmoidImpl::forward(const Tensor& input) {
+  return sigmoid(input);
```
nit: would be great to write:
```suggestion
  return torch::sigmoid(input);
```
to strictly match Python version :D

pbelevich committed:
Update on "C++ API parity: Sigmoid"


Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"


Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"


Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"


Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"


Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"


Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"


Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"


Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

[ghstack-poisoned]

yf225 reviewed (approved):
Thanks a lot @pbelevich!

pbelevich committed:
Update on "C++ API parity: Sigmoid"


Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"


Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"


Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: Sigmoid"


Differential Revision: [D17835405](https://our.internmc.facebook.com/intern/diff/D17835405)

[ghstack-poisoned]
